src/error.ts

```diff
@@ -0,0 +1,21 @@
+export class InputError extends Error {
+  code: number;
+  additionalInfo?: string;
+
+  constructor(code: number, message: string, additionalInfo?: string) {
+    super();
+
+    this.message = message;
+    this.code = code;
+    this.additionalInfo = additionalInfo ?? null;
+  }
+}
+
+export const errorsMessages: { existingEmail: string; weakPassword: string } = {
```

Comment:
a tipagem aqui ta mais para um Record
```suggestion
export const errorsMessages: Record<string, string> = {
```

Se quiser fazer um enum para as chaves desse objeto para nao precisar tipar o primeiro argumento como apenas string...